### PR TITLE
Fix HTTP attributes for deps, fix success and response/resultCode for Azure Exporter

### PR DESCRIPTION
**Pull request by lzchen**
Address [#825] and [#826], [#831] and [#832]

Adds "component" span attributes to `requests` and `httplib` integrations to specify "http" type.
Adds logic to popular success and responseCode based off span status or status_code in span attributes.

### CODE REVIEW

**Comment by lzchen** [comment]
@lmolkova 
Please review if you have the time. It consists of the same changes as for OpenTelemetry.

File: contrib/opencensus-ext-azure/opencensus/ext/azure/trace_exporter/__init__.py [hunk 1]
```diff
@@ -87,32 +92,45 @@ def span_data_to_envelope(self, sd):
...
+                        # name is METHOD/path
+                        data.name = sd.attributes['http.method'] \
+                            + "/" + parse_url.path
                 if 'http.status_code' in sd.attributes:
-                    data.resultCode = str(sd.attributes['http.status_code'])
+                    status_code = sd.attributes["http.status_code"]
+                    data.resultCode = str(status_code)
+                    data.success = 200 <= status_code < 400
+                elif sd.status.code == 0:
+                    data.success = True
```
Comment (lmolkova) [inline contrib/opencensus-ext-azure/opencensus/ext/azure/trace_exporter/__init__.py]: what if it's not SERVER or CLIENT span (Unspecified), we create Dependency and we should still set status

File: contrib/opencensus-ext-azure/opencensus/ext/azure/trace_exporter/__init__.py [hunk 2]
```diff
@@ -55,22 +60,22 @@ def span_data_to_envelope(self, sd):
...
+                id='{}'.format(sd.span_id),
                 duration=utils.timestamp_to_duration(
                     sd.start_time,
                     sd.end_time,
                 ),
-                responseCode='0',
-                success=False,
+                responseCode=str(sd.status.code),
+                success=False,  # Modify based off attributes or status
                 properties={},
```
Comment (lmolkova) [inline contrib/opencensus-ext-azure/opencensus/ext/azure/trace_exporter/__init__.py]: this was not listed as a bug, but I noticed we don't populate properties for HTTP requests. Can we do it?

request.name = "METHOD /route" (if route is not populated, then path)
request.url = url

**Comment by lzchen** [comment]
@lmolkova 
I have made the changes you've suggested!

**Pull request merged**